### DISCUSSION TITLE
Conflicting plugins

### DISCUSSION THREAD

**Discussion by Macweese** [runelite/runelite#16241] [Ideas]
I wasn't sure whether or not this should be an issue opened to RuneLite or respective plugin providers since the issue is split equally.
If multiple plugins share properties unexpected behavior may occur. 

Background
After installing plugins (from different developers) and noticing they were misbehaving, I determined it to be because they shared the same name.

Example: both share `TombsOfAmascutPlugin`
[Plugin A](https://github.com/LlemonDuck/tombs-of-amascut/tree/master/src/main/java/com/duckblade/osrs/toa)
[Plugin B](https://github.com/BoydHarold/Tombs-of-Amascut-Puzzle-Helper/tree/cdabb2b398a88cc75895863341fce8c5088b9b2d/src/main/java/com/tombsofamascutpuzzlehelper)

Is this something RuneLite would/should control? Or would it be the responsibility of the plugin provider to ensure their plugin does not conflict with existing plugins, from the plugin-hub. While PluginDescriptor has `conflicts` it requires both/all providers to coordinate together and converge on a common key. Unfortunately I doubt this feature will see itself being used, an idea could be to have itself added by default...? If not for the PluginManager to handle these cases, perhaps adding a caveat to inform developers in the [README](https://github.com/runelite/plugin-hub#using-the-template-repository).

Have not confirmed, although suspect this may also be true for configs.

The linked repositories are one example of conflicts, there may be more — I have not checked.

**Comment by Nightfirecat** [comment]
This is somewhat of a known issue. Class names, along with several other properties (most notably config group keys) are not namespaced on the plugin hub, so conflicts like this can occur. That said, it is something we would like to eventually fix.
